samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/author/PublishAssessmentListener.java

```diff
@@ -145,15 +145,14 @@ public PublishAssessmentListener() {
   }
 
   public void processAction(ActionEvent ae) throws AbortProcessingException {
+
 	  repeatedPublishLock.lock();
-	  try {
 
+	  try {
   		//FacesContext context = FacesContext.getCurrentInstance();
-  		if (ae == null) {
-  			repeatedPublish = false;
-  			return;
-  		}
-  		else {
```

Comment:
@adrianfish i've been trying to debug here and it's been super difficult. the old code is obviously insane!  but after this change, just using the Publish from the action dropdown for one assessment publishes the assessment (as opposed to waiting for the next screen). do you remember anything here?

maybe getting rid of the single publish option makes sense. idk. just trying to figure out right way

Reply:
I don't remember why I removed those lines tbh. Maybe I was just trying to align it with the assignments workflow. What's the issue? What problem has this caused? I'm out of the loop a bit as I've been away.

Reply:
My instinct is you removed them because of the new bulk publishing workflow. the problem now is that the single publish (from the dropdown) option does an immediate publish ... breaking the notification workflow

Reply:
@ern and I talked about this a bit. it seems like we can edit samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/author/ActionSelectListener.java and have it send an ActionEvent over to this method so we can determine where the publish is coming from